CMakeLists.txt

```diff
@@ -80,19 +79,25 @@ target_include_directories(mbed-os
         ${CMAKE_CURRENT_SOURCE_DIR}
 )
 
-# Default build
 add_subdirectory(cmsis)
-add_subdirectory(components)
-add_subdirectory(connectivity)
 add_subdirectory(drivers)
 add_subdirectory(events)
-add_subdirectory(features)
 add_subdirectory(hal)
 add_subdirectory(platform)
 add_subdirectory(rtos)
-add_subdirectory(storage)
 add_subdirectory(targets)
 
+# These components are optional (using EXCLUDE_FROM_ALL)
+add_subdirectory(connectivity EXCLUDE_FROM_ALL)
+add_subdirectory(storage EXCLUDE_FROM_ALL)
+
+# Device key depends on storage, optional
+add_subdirectory(drivers/device_key EXCLUDE_FROM_ALL)
+# USB component, optional
+add_subdirectory(drivers/source/usb EXCLUDE_FROM_ALL)
+
+#add_subdirectory(features EXCLUDE_FROM_ALL)
```

Comment:
Can we remove these instead of commenting them out?

Reply:
I can if we can't fix it soon